protocol/xmrmaker/net.go

```
@@ -144,18 +146,25 @@ func (inst *Instance) HandleInitiateMessage(
 		offer.EthAsset,
 	)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	state, err := inst.initiate(takerPeerID, offer, offerExtra, providedPiconero, expectedAmount)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	if err = state.handleSendKeysMessage(msg); err != nil {
-		return nil, nil, err
+		_ = state.Exit()
+		return nil, err
 	}
 
 	resp := state.SendKeysMessage()
-	return state, resp, nil
+	err = inst.backend.SendSwapMessage(resp, offer.ID)
+	if err != nil {
+		_ = state.Exit()
```

Whatever solution you use on the other swallowed error should be here too.